pycket/pycket-lang/zoTransform.rkt

```diff
@@ -0,0 +1,581 @@
+         [args-part (apply-values-args-expr app-form)]
+
+         [mod-var (if (not (toplevel? proc-part))
+                      (error 'handle-apply-values "proc is NOT a toplevel")
+                      (list-ref toplevels (toplevel-pos proc-part)))]
+
+         [proc (if (not (module-variable? mod-var))
+                   (error 'handle-apply-values "toplevel is NOT a module-variable")
+                   (to-ast-single mod-var toplevels localref-stack))]
+
+         [args (to-ast-single args-part toplevels localref-stack)]
+
+         ;; construct the lam (lambda () args)
+         [lambda-form (hash* 'source (hash* '%p (string-append pycketDir "fromBytecode_" moduleName ".rkt")) ;; toplevel application
+                             'position 321
+                             'span 123
+                             'module (hash* '%mpi (hash* '%p (path->string
+                                                              (resolved-module-path-name
+                                                               (module-path-index-resolve
+                                                                (module-variable-modidx mod-var))))))
+                             'lambda '()
+                             'body (list args))]
+         )
+
+  ;; (call-with-values lam proc)
+    (hash* 'operator (hash* 'source-name "call-with-values")
+           'operands (list lambda-form proc))
+    ))
+
+(define (handle-localref lref-form toplevels localref-stack)
+  (let* (
+         (pos (localref-pos lref-form))
+         (stack-slot-raw (list-ref localref-stack pos))
+         (stack-slot (if (box? stack-slot-raw) (unbox stack-slot-raw) stack-slot-raw))
+         )
+    (cond
+      [(hash? stack-slot) stack-slot]
+      [else (hash* 'lexical stack-slot)])))
+
+;;(define (handle-toplevel toplevel-form toplevels)
+  ;;(let ((pos (toplevel-pos toplevel-form)))
+
+(define (handle-module-variable mod-var toplevels localref-stack)
+  (let ((name (symbol->string (module-variable-sym mod-var)))
+        (module-path (path->string
+                         (resolved-module-path-name
+                          (module-path-index-resolve
+                           (module-variable-modidx mod-var))))))
+  (hash* 'source-name name
+         'source-module (list module-path))))
+
+(define (handle-let-one letform toplevels localref-stack)
+  (let* (
+         (newstack-prev (cons 'let-one-uninitialized-slot localref-stack)) ;; push uninitialized slot
+         (rhs (to-ast-single (let-one-rhs letform) toplevels newstack-prev)) ;; evaluate rhs
+         (newstack (cons rhs (cdr newstack-prev))) ;; put the rhs to the slot
+        )
+    (hash* 'let-bindings '()
+           'let-body (list (to-ast-single (let-one-body letform)
+                                          toplevels
+                                          newstack)))))
+
+(define (body-name body-form)
+  (cond
+    ((list? body-form) "ATTENTION : we have a LIST")
```

Comment:
Well it seemed to me at first that we don't get a list as a single byte-form, but somewhere in my tests I noticed we can have that, but I still don't know exactly the cases where that happens. I put it there to easily identify those cases until I figure out a way to properly handle them.